CIP-0095/README.md

```diff
@@ -0,0 +1,678 @@
+### Full Governance API
+
+Methods contained in this section on invocation should request the user to
+review and to consent to signature and submission. The exceptions to this are
+`.getActiveStakeKeys()` and `.getDRepKey()`, user consent should not be needed
+to share public key information.
+
+#### `api.getDRepKey(): Promise<PubDRepKey>`
+
+Errors: `APIError`
+
+The connected wallet account provides the account's public DRep Key, derivation
+as described in [DRep Key](#DRep-key).
+
+##### Returns
+
+The wallet account's DRep Key.
+
+#### `api.getActiveStakeKeys(): Promise<PubStakeKey[]>`
+
+Errors: `APIError`
+
+The connected wallet account's active public stake keys (with keys which are
+being used for staking), if the wallet tracks the keys that are used for
+governance then only those keys shall be returned.
+
+These are used by the client to identify the user's on-chain CIP-1694
+interactions. Here we allow for multiple stake credentials to provided at once
+for the case of
+[multi-stake key wallets](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0018).
+
+##### Returns
+
+An array of the connected user's active stake keys.
+
+#### `api.submitVoteDelegation([tx: UnsignedTransaction, stakeKey: PubStakeKey]): Promise<SubmittedTransaction[]>`
+
+Errors: [APIError](#extended-apierror), [`TxSignError`](#extended-txsignerror)
+
+This endpoint requests the wallet to inspect, sign and submit transaction(s)
+containing vote delegation certificates. The wallet should articulate this
+request from client application in a explicit and highly informative way. Users
+must be shown the target of the delegation (DRepID or a predefined DRep
+identifier) and must be informed which of their stake keys are being used. For
+the case of multiple delegations at once, the user may only be asked for
+permission to authorize all at once, but this can depend on wallet's
+implementation.
+
+If user grants permission, each transaction must be signed by the secret key of
+the provided public stake key, with the signature and key to be added to the
+transaction witness set before submission.
+
+By allowing clients to supply the stake key we are placing the burden of
+"multi-governance-delegation" management onto the client, reducing the
+complexity for wallet implementations. By forcing wallets to inspect these
+certificates it allows the user to catch malicious client applications
+attempting to insert their own delegation targets.
+
+##### Errors
+
+One `TxSignError` should be returned if there is a signature error with any of
+the certificates.
+
+##### Returns
+
+This returns an array of `SubmittedTransaction` objects which contain the
+details of the submitted delegation certificates, for the client to confirm. The
+returned `txHash`s can be used by the client to track the status of the
+transactions containing the certificates on Cardano.
+
+#### `api.submitDRepRegistration(tx: UnsignedTransaction, dRepKey: PubDRepKey): Promise<SubmittedTransaction>`
```

Comment:
🎨 Given that a wallet have only one drep key, could you make `dRepKey` optional ?
When null, the wallet would automatically use the one drep key or return an error if none exists.

The same would apply to `submitDRepRetirement()` and `submitVote()`

Reply:
Making `dRepkey` optional or even removing it seems sensible, I will amend. The reason I kept it there was just to match the flow of Vote Delegation (pass transaction and tell wallet exactly what key to use).